src/python/pants/backend/tools/taplo/subsystem.py

```diff
@@ -0,0 +1,92 @@
+from pants.option.option_types import ArgsListOption, BoolOption, SkipOption, StrListOption
+from pants.source.filespec import FilespecMatcher
+from pants.util.strutil import softwrap
+
+
+class Taplo(TemplatedExternalTool):
+    help = "An autoformatter for TOML files (https://taplo.tamasfe.dev/)"
+
+    options_scope = "taplo"
+    name = "Taplo"
+    default_version = "0.8.0"
+    default_known_versions = [
+        "0.8.0|macos_arm64|79c1691c3c46be981fa0cec930ec9a6d6c4ffd27272d37d1885514ce59bd8ccf|3661689",
+        "0.8.0|macos_x86_64|a1917f1b9168cb4f7d579422dcdf9c733028d873963d8fa3a6f499e41719c502|3926263",
+        "0.8.0|linux_arm64|a6a94482f125c21090593f94cad23df099c4924f5b9620cda4a8653527c097a1|3995383",
+        "0.8.0|linux_x86_64|3703294fac37ca9a9f76308f9f98c3939ccb7588f8972acec68a48d7a10d8ee5|4123593",
+    ]
+    default_url_template = (
+        "https://github.com/tamasfe/taplo/releases/download/{version}/taplo-{platform}.gz"
+    )
+    default_url_platform_mapping = {
+        "macos_arm64": "darwin-aarch64",
+        "macos_x86_64": "darwin-x86_64",
+        "linux_arm64": "linux-aarch64",
+        "linux_x86_64": "linux-x86_64",
+    }
+
+    skip = SkipOption("fmt", "lint")
+    args = ArgsListOption(example="--option align_entries=false")
+    config_discovery = BoolOption(
+        default=True,
+        advanced=True,
+        help=softwrap(
+            """
+            If true, Pants will include all relevant `taplo.toml` or `.taplo.toml`
+            files during a run.
+            """
+        ),
+    )
+
+    include = StrListOption(
+        help=softwrap(
+            """
+            A list of glob patterns of files to include in formatting relative to the
+            build root.
+
+            Example:
+
+                ["**/*.toml", "**/pyproject.toml"]
+            """
+        ),
+    )
+    exclude = StrListOption(
+        help=softwrap(
+            """
+            A list of glob patterns of files to exclude from formatting relative to the
+            build root.
+
+            Example:
+
+                ["src/*.toml", "pants/pyproject.toml"]
+            """
+        ),
+    )
+
+    def generate_exe(self, plat: Platform) -> str:
+        return f"./{self.generate_url(plat).rsplit('/', 1)[-1].removesuffix('.gz')}"
+
+    def config_request(self, dirs: Sequence[str]) -> ConfigFilesRequest:
+        candidates = [os.path.join(d, ".taplo.toml") for d in ("", *dirs)]
+        candidates.extend(os.path.join(d, "taplo.toml") for d in ("", *dirs))
+        return ConfigFilesRequest(
+            discovery=self.config_discovery,
+            check_existence=candidates,
+        )
+
+    def filter_inputs(self, filepaths: Sequence[str]) -> dict[str, str]:
+        """Returns a mapping of path to path."""
+        matched_filepaths = FilespecMatcher(includes=self.includes, excludes=self.excludes).matches(
+            filepaths
+        )
+        return {m: m for m in matched_filepaths}
```

Comment:
This returns a dictionary they keys map to themselves? That's odd.
Also there's only one caller.

I think you can move this code into the partition rule. and skip the method/intermediate dictionary.